antenna/util.py

```diff
@@ -211,25 +206,51 @@ def _retry_inner(fun):
         @wraps(fun)
         def _retry_fun(*args, **kwargs):
             attempts = 0
+            wait_times = wait_time_generator()
+
             while True:
+                # Grab the next wait time and if there isn't one, then this is
+                # the last attempt
+                try:
+                    next_wait = next(wait_times)
+                except StopIteration:
+                    next_wait = None
+
                 try:
                     ret = fun(*args, **kwargs)
                     if retryable_return is None or not retryable_return(ret):
+                        # This was a successful return--yay!
                         return ret
 
-                    elif max_attempts is not None and attempts >= max_attempts:
-                        raise MaxAttemptsError(
-                            'Maximum retry attempts. Error return.', ret
+                    # The return value is "bad", so we log something and then
+                    # do another iteration.
+                    if module_logger is not None:
+                        module_logger.warning(
+                            '%s: bad return, retry attempt %s',
+                            fun.__qualname__,
+                            attempts
```

Comment:
I added logging here because it occurred to me we'd like some heads-up when things are going wonky regarding returned results.